File: unit/unit/src/main/java/com/cedarsoft/unit/si/ns.java
package com.cedarsoft.unit.si;

import java.lang.annotation.Documented;
import java.lang.annotation.Inherited;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import com.cedarsoft.unit.Name;
import com.cedarsoft.unit.Symbol;
import com.cedarsoft.unit.Unit;
import com.cedarsoft.unit.prefix.milli;
import com.cedarsoft.unit.prefix.nano;
import com.cedarsoft.unit.quantity.Time;

/**
 * @author Johannes Schneider (<a href="mailto:js@cedarsoft.com">js@cedarsoft.com</a>)
 */
@Retention( RetentionPolicy.RUNTIME )
@Documented
@Inherited

@Unit
@Time
@Symbol( ns.SYMBOL )
@Name( "nanoseconds" )
@nano( s.class )
public @interface ns {
  String SYMBOL = "ns";
}

File: unit/unit/src/commonMain/kotlin/com/cedarsoft/unit/si/ns.kt
/**
 * Copyright (C) cedarsoft GmbH.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.cedarsoft.unit.si

import com.cedarsoft.unit.Name
import com.cedarsoft.unit.Symbol
import com.cedarsoft.unit.Unit
import com.cedarsoft.unit.prefix.nano
import com.cedarsoft.unit.quantity.Time

/**
 * @author Johannes Schneider ([js@cedarsoft.com](mailto:js@cedarsoft.com))
 */
@Retention(AnnotationRetention.RUNTIME)
@MustBeDocumented
@Suppress("ClassName")
@Unit
@Time
@Symbol(ns.SYMBOL)
@Name("nanoseconds")
@nano(s::class)
annotation class ns {
  companion object {
    const val SYMBOL: String = "ns"
  }
}

